Tests/SourceKitTests/ImplementationTests.swift

```diff
@@ -0,0 +1,67 @@
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@testable import SourceKit
+import LanguageServerProtocol
+import XCTest
+import SKTestSupport
+import ISDBTestSupport
+
+final class ImplementationTests: XCTestCase {
+  func testImplementation() throws {
+    let ws = try staticSourceKitTibsWorkspace(name: "Implementation")!
+    try ws.buildAndIndex()
+
+    try ws.openDocument(ws.testLoc("a.swift").url, language: .swift)
+    try ws.openDocument(ws.testLoc("b.swift").url, language: .swift)
+
+    func impls(at testLoc: TestLocation) throws -> [Location] {
+      let textDocument = TextDocumentIdentifier(testLoc.url)
+      let request = ImplementationRequest(textDocument: textDocument, position: Position(testLoc))
+      let implementations = try ws.sk.sendSync(request)
+      return implementations
+    }
+    func testLoc(_ name: String) -> TestLocation {
+      ws.testLoc(name)
+    }  
+    func loc(_ name: String) -> Location {
+      Location(ws.testLoc(name))
+    }
+    
+    try XCTAssertEqual(impls(at: testLoc("Protocol")), [loc("StructConformance")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolStaticVar")), [loc("StructStaticVar")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolStaticFunction")), [loc("StructStaticFunction")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolVariable")), [loc("StructVariable")])
+    try XCTAssertEqual(impls(at: testLoc("ProtocolFunction")), [loc("StructFunction")])
+    try XCTAssertEqual(impls(at: testLoc("Class")), [loc("SubclassConformance")])
+    try XCTAssertEqual(impls(at: testLoc("ClassClassVar")), [loc("SubclassClassVar")])
+    try XCTAssertEqual(impls(at: testLoc("ClassClassFunction")), [loc("SubclassClassFunction")])
+    try XCTAssertEqual(impls(at: testLoc("ClassVariable")), [loc("SubclassVariable")])
+    try XCTAssertEqual(impls(at: testLoc("ClassFunction")), [loc("SubclassFunction")])
+
+    try XCTAssertEqual(impls(at: testLoc("Sepulcidae")), [loc("ParapamphiliinaeConformance"), loc("XyelulinaeConformance"), loc("TrematothoracinaeConformance")])
+    try XCTAssertEqual(impls(at: testLoc("Parapamphiliinae")), [loc("MicramphiliusConformance"), loc("PamparaphiliusConformance")])
+    try XCTAssertEqual(impls(at: testLoc("Xyelulinae")), [loc("XyelulaConformance")])
+    try XCTAssertEqual(impls(at: testLoc("Trematothoracinae")), [])
+
+    try XCTAssertEqual(impls(at: testLoc("Prozaiczne")), [loc("MurkwiaConformance2"), loc("SepulkaConformance1")])
+    // FIXME: non-ascii characters break Tibs `TestLocation`s
```

Comment:
What is the issue here?

Reply:
I got confused by late night debugging. There is a tibs bug, but I accidentally dodged it here, because "ć" and "Ł" fit into a single utf-16 code unit, even though they aren't ascii. The bug that the test caught is real though and needs investigating.

About the tibs bug - if I put this in a test file
```
class 😃 {}/*a*/
class B {}/*b*/
class Ć {}/*c*/
class 👩‍👩‍👧‍👧 {}/*d*/
```
and then `print(loc("a").range.lowerBound.utf16index)`, I get 15 for all of them, which suggests that tibs counts Characters instead of utf-16 code units. I think I should get 16, 15, 15, 25 for the column numbers

Reply:
Ah, thanks! I'll fix it.

Reply:
Okay, I think I fixed this in 
https://github.com/apple/indexstore-db/pull/38
https://github.com/apple/sourcekit-lsp/pull/136

Unfortunately, this will probably expose the latent problem that we are using a utf8 index from the `SymbolOccurrence`, so your comparisons may start failing.  If that's the case, I suggest adding new initializers to `Position(badUTF16: TestLocation)` and similarly for `Location` that do the same incorrect behaviour as we do in the implementation.  It also makes it easier to find the places that should change when we someday fix the underlying utf8 vs. utf16 issue from our index data.

Reply:
Actually, assuming the tests pass we could just land your changes first, and I can fix up any discrepancies in my PR.